modules/identity/src/Volo.Abp.Identity.Domain/Volo/Abp/Identity/IdentitySecurityLogStore.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Volo.Abp.DependencyInjection;
+using Volo.Abp.Guids;
+using Volo.Abp.SecurityLog;
+using Volo.Abp.Uow;
+
+namespace Volo.Abp.Identity
+{
+    [Dependency(ReplaceServices = true)]
+    public class IdentitySecurityLogStore : ISecurityLogStore, ITransientDependency
+    {
+        public ILogger<IdentitySecurityLogStore> Logger { get; set; }
+
+        protected AbpSecurityLogOptions SecurityLogOptions { get; }
+        protected IIdentitySecurityLogRepository IdentitySecurityLogRepository { get; }
+        protected IGuidGenerator GuidGenerator { get; }
+        protected IUnitOfWorkManager UnitOfWorkManager { get; }
+
+        public IdentitySecurityLogStore(
+            ILogger<IdentitySecurityLogStore> logger,
+            IOptions<AbpSecurityLogOptions> securityLogOptions,
+            IIdentitySecurityLogRepository identitySecurityLogRepository,
+            IGuidGenerator guidGenerator,
+            IUnitOfWorkManager unitOfWorkManager)
+        {
+            Logger = logger;
+            SecurityLogOptions = securityLogOptions.Value;
+            IdentitySecurityLogRepository = identitySecurityLogRepository;
+            GuidGenerator = guidGenerator;
+            UnitOfWorkManager = unitOfWorkManager;
+        }
+
+        public async Task SaveAsync(SecurityLogInfo securityLogInfo)
+        {
+            if (!SecurityLogOptions.IsEnabled)
+            {
+                return;
+            }
+
+            using (var uow = UnitOfWorkManager.Begin(requiresNew: true))
```

Comment:
This can also be a "non transactional" UOW since it contains only one insert. It will reduce the dedlock probability.

Reply:
`Begin(requiresNew: true)` is extension method. https://github.com/abpframework/abp/blob/ae8095d1eac23cad4d7f652012578157a5ba5a86/framework/src/Volo.Abp.Uow/Volo/Abp/Uow/UnitOfWorkManagerExtensions.cs#L10-L16